engine/src/main/java/org/terasology/rendering/nui/internal/NUIManagerInternal.java

```diff
@@ -65,7 +70,7 @@
  */
 public class NUIManagerInternal extends BaseComponentSystem implements NUIManager {
     private Logger logger = LoggerFactory.getLogger(NUIManagerInternal.class);
-    private Deque<UIScreenLayer> screens = Queues.newArrayDeque();
+    private LinkedList<UIScreenLayer> screens = new LinkedList<>();
```

Comment:
The list interface is needed to [reload individual screens](https://github.com/MovingBlocks/Terasology/pull/1896/files#diff-c6ee4291cb4c4614c5d299b8fe8656e7R567).